java/ql/src/semmle/code/java/StringFormat.qll

```diff
@@ -449,4 +484,9 @@ private class LoggerFormatString extends FormatString {
   }
 
   override int getMaxFmtSpecIndex() { result = count(int i | fmtPlaceholder(i)) }
+
+  override int getAnArgUsageOffset(int argNo) {
+    fmtPlaceholder(result) and
+    argNo = count(int i | fmtPlaceholder(i) and i < result)
```

Comment:
This is needlessly quadratic, and also off-by-one - `argNo` is 1-based, but calculated base 0 here. This change should fix both issues:
```suggestion
    result = rank[argNo](int i | fmtPlaceHolder(i))
```